src/Controller/RegistrationController.php

```diff
@@ -62,7 +62,13 @@ public function registerAction(Request $request, string $voucher = null): Respon
                 }
 
                 $recoveryToken = $user->getPlainRecoveryToken();
-                $user->erasePlainRecoveryToken();
+                if (null === $recoveryToken) {
+                    throw new \Exception('recoveryToken should not be null');
```

Comment:
I don't fully get why you added this, probably a leftover from testing? What condition do you have in mind that might lead to `null === $recoveryToken` being true now that you fixed the bug? If it's merely to catch the bug, I would suggest to remove it instead.

Also you could merge the assignment and condition:

```php
if (null === $recoveryToken = $user->getPlainRecoveryToken()) {
   ...
}
```